lib/betagouvbot/webhook.rb

```diff
@@ -52,20 +52,38 @@ def members
     end
 
     post '/compte' do
-      member   = params['text'].to_s.split.first
-      origin   = params['user_name']
-      response = "A la demande de @#{origin} je créée un compte pour #{member}"
-      body     = { response_type: 'in_channel', text: response }.to_json
-      HTTParty.post(params['response_url'], body: body, headers: headers)
-
-      response = 'OK, création de compte en cours !'
-      accounts = AccountRequest.(members, *params['text'].to_s.split)
-      execute  = params.key?('token') && (params['token'] == ENV['COMPTE_TOKEN'])
-      accounts.map(&:execute) if execute
-
-      response = 'Je ne vois pas de qui tu veux parler' if accounts.empty?
-      body     = { text: response }.to_json
-      HTTParty.post(params['response_url'], body: body, headers: headers)
+      member, email, password = params['text'].to_s.split
+      account_request         = AccountRequest.new(members, member, email, password)
+
+      account_request.on(:success) do |accounts|
+        # Notify request is valid and being treated...
+        origin   = params['user_name']
+        response = "A la demande de @#{origin} je créée un compte pour #{member}"
+        body     = { response_type: 'in_channel', text: response }.to_json
+        HTTParty.post(params['response_url'], body: body, headers: headers)
+
+        execute = params.key?('token') && (params['token'] == ENV['COMPTE_TOKEN'])
+        accounts.map(&:execute) if execute
```

Comment:
We might not need this to go live, but I just remembered that the API can still raise if the password is too short. Perhaps we can just add a length condition to validate_password, then (as long as the length condition remains the same on the other end) we guarantee the exception will not be triggered. Otherwise we have to catch and handle. WDYT?